### PR TITLE
Upgrade to Ruby 3.x

### PR DESCRIPTION
**Pull request by qwcode**
* Upgrades the [head](https://github.com/ThoughtWorksStudios/eb_deployer/commits/master) of the thoughtworks clone to Ruby 3.x
* changes the name to `man_eb_deployer`
* adds Tests and Release GHA workflows.

### CODE REVIEW

File: eb_deployer.gemspec
Comment (duckpuppy) [inline eb_deployer.gemspec:1]: The e-mail addresses and homepage in the gemspec are all from 5 years ago and should also be updated.

**Review by duckpuppy — CHANGES_REQUESTED** [review]

**Review by duckpuppy — APPROVED** [review]

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  pull_request: {}
+  workflow_dispatch: {}
+
+jobs:
+  test:
+    name: Unit Tests
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - uses: ruby/setup-ruby@52b8784594ec115fd17094752708121dc5dabb47
```
Comment (jeremiahlukus) [inline .github/workflows/test.yml:13]:
```
   - uses: ruby/setup-ruby@52b8784594ec115fd17094752708121dc5dabb47    
      with:
        bundler-cache: true
 ```
 
 I think this is faster than doing a run buddle install

Comment (qwcode) [reply .github/workflows/test.yml:13]: ok, updated

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+name: Release
+on:
+  workflow_dispatch: {}
+  push:
+    branches:
+      - main
```
Comment (codezninja) [inline .github/workflows/release.yml]:
```suggestion
      - master
```

unless we are planning on updating the repo to main

**Review by codezninja — APPROVED** [review]
lgtm. Just had that one workflow change unless we plan on updating to main

File: .github/workflows/release.yml (hunk 2 above)
Comment (qwcode) [reply .github/workflows/release.yml]: switched to master for now

**Pull request merged**